request/middleware.py

```diff
@@ -11,35 +12,60 @@
 logger = logging.getLogger('request.security.middleware')
 
 
-class RequestMiddleware(MiddlewareMixin):
-    def process_response(self, request, response):
+class RequestMiddleware:
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        start_time = time.monotonic()
+        response = self.get_response(request)
+        end_time = time.monotonic()
+        response_time = timedelta(end_time - start_time)
```

Comment:
```suggestion
        response_time = timedelta(seconds=end_time - start_time)
```

(otherwise the first argument is days !!)